vllm/platforms/xpu.py

```diff
@@ -34,3 +41,17 @@ def get_device_total_memory(cls, device_id: int = 0) -> int:
     @staticmethod
     def inference_mode():
         return torch.no_grad()
+
+    @classmethod
+    def check_and_update_config(cls, vllm_config: VllmConfig) -> None:
+        # check and update model config
+        model_config = vllm_config.model_config
+        if model_config.dtype == torch.bfloat16:
+            logger.warning(
+                "bfloat16 is not fully supported on XPU, casting to float16.")
+            model_config.dtype = torch.float16
+        if not model_config.enforce_eager:
+            logger.warning(
+                "CUDA graph is not supported on XPU, fallback to the eager "
+                "mode.")
+            model_config.enforce_eager = True
```

Comment:
seems lost the original `parallel_config` check?
```
    if (config.distributed_executor_backend is not None
            and config.distributed_executor_backend != "ray"):
        logger.warning(
            "%s is not supported on XPU, fallback to ray distributed executor "
            "backend.", config.distributed_executor_backend)
        config.distributed_executor_backend = "ray"
```

Reply:
I think that function `_verify_and_get_parallel_config` is not called.

Reply:
This check should be included for xpu platform. The call stack seems get broken in [commit](https://github.com/vllm-project/vllm/commit/18bd7587b78b3b9868fea29d59ae8c3600c3e5a5#diff-d1c5ec4ddd588e3b7cac13bde85a98ac5b20686dc16b9da3b1c324467c3be2b5).

Reply:
oh sorry for that. can you help review https://github.com/vllm-project/vllm/pull/10479 ?